pkg/gsheets/sheet.go

```diff
@@ -0,0 +1,185 @@
+package gsheets
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/shiftstack/gazelle/pkg/job"
+
+	"golang.org/x/net/context"
+	"google.golang.org/api/sheets/v4"
+)
+
+const offset int = 6
+
+type Sheet struct {
+	JobName string
+	Client  *Client
+
+	idxList []int
+}
+
+func (s *Sheet) getID() int64 {
+	innerMap := map[string]int64{
+		"release-openshift-ocp-installer-e2e-openstack-4.4":           552238361,
+		"release-openshift-ocp-installer-e2e-openstack-serial-4.4":    923764376,
+		"release-openshift-ocp-installer-e2e-openstack-4.3":           1408408210,
+		"release-openshift-ocp-installer-e2e-openstack-serial-4.3":    1992493886,
+		"release-openshift-ocp-installer-e2e-openstack-4.2":           493400895,
+		"release-openshift-ocp-installer-e2e-openstack-serial-4.2":    254644539,
+		"release-openshift-origin-installer-e2e-openstack-4.2":        61282267,
+		"release-openshift-origin-installer-e2e-openstack-serial-4.2": 1799792403,
+	}
+
+	sheetId, ok := innerMap[s.JobName]
+	if !ok {
+		log.Fatalf("Unknown job %v", s.JobName)
+	}
+
+	return sheetId
+}
+
+func (s *Sheet) getName() string {
+	innerMap := map[string]string{
+		"release-openshift-ocp-installer-e2e-openstack-4.4":           "Parallel OCP 4.4",
+		"release-openshift-ocp-installer-e2e-openstack-serial-4.4":    "Serial OCP 4.4",
+		"release-openshift-ocp-installer-e2e-openstack-4.3":           "Parallel OCP 4.3",
+		"release-openshift-ocp-installer-e2e-openstack-serial-4.3":    "Serial OCP 4.3",
+		"release-openshift-ocp-installer-e2e-openstack-4.2":           "Parallel OCP 4.2",
+		"release-openshift-ocp-installer-e2e-openstack-serial-4.2":    "Serial OCP 4.2",
+		"release-openshift-origin-installer-e2e-openstack-4.2":        "Parallel Origin 4.2",
+		"release-openshift-origin-installer-e2e-openstack-serial-4.2": "Serial Origin 4.2",
+	}
+
+	sheetName, ok := innerMap[s.JobName]
+	if !ok {
+		log.Fatalf("Unknown job %v", s.JobName)
+	}
+
+	return sheetName
+}
+
+func (s *Sheet) getAllIDs() {
```

Comment:
why do we need all the IDs? Is that for performing insertions in the middle? Can't we just limit ourselves to appending to the top?

Reply:
For a two reasons: you can now set the range in whatever order suits you, 405-407 or 407-405, and you'll get consistent result in the sheet, or pass gazelle the ID of old jobs you want to refresh the results.

Reply:
wouldn't be cool if we started from the top :P